src/core/integrators/stokesian_dynamics_inline.hpp

```diff
@@ -37,12 +37,12 @@ inline void stokesian_dynamics_propagate_vel_pos(const ParticleRange &particles,
   for (auto &p : particles) {
 
     // Translate particle
-    p.r.p += p.m.v * time_step;
+    p.pos() += p.v() * time_step;
 
     // Perform rotation
-    auto const norm = p.m.omega.norm();
-    if (norm != 0) {
-      auto const omega_unit = (1. / norm) * p.m.omega;
+    auto const norm = p.omega().norm();
+    if (norm != 0.) {
+      auto const omega_unit = (1. / norm) * p.omega();
```

Comment:
is there normalize() for our Vectors? If not would be nice to have

Reply:
The method exists, but then we are computing the norm twice: once to decide whether it is zero, and once to normalize the vector. Clang 13.0 and GCC 11.2 do not realize that the same value can be re-used, even at `-O3`. Since this is a hot loop, performance is more important than readability.